File: src/main/resources/output/IFC4X3_python/Types/IfcVectorOrDirection.py
# SELECT TYPE IfcVectorOrDirection
# Options:
# - IfcDirection
# - IfcVector
class IfcVectorOrDirection:
    def __init__(self, value):
        if not isinstance(value, (IfcDirection, IfcVector)):
            raise TypeError("value must be one of: IfcDirection, IfcVector")
        self.value = value

    def get_value_as(self, cls):
        return cls(self.value) if isinstance(self.value, cls) else None

File: src/main/resources/output/IFC4X3_csharp/Types/IfcVectorOrDirection.cs
// SELECT TYPE IfcVectorOrDirection
public class IfcVectorOrDirection {
    public object Value { get; }

    public IfcVectorOrDirection(object value) {
        if (!(value is IfcDirection || value is IfcVector))
        {
            throw new ArgumentException("Value must be one of: IfcDirection, IfcVector");
        }
        Value = value;
    }

    public T GetValueAs<T>() {
        return (T)Value;
    }
}
